src/week2.hs

```diff
@@ -0,0 +1,168 @@
+{-# OPTIONS_GHC -Wall #-}
+--module week2 where
+
+-- Mastermind -----------------------------------------
+
+-- A peg can be one of six colors
+data Peg = Red | Green | Blue | Yellow | Orange | Purple
+         deriving (Show, Eq, Ord)
+
+-- A code is defined to simply be a list of Pegs
+type Code = [Peg]
+
+-- A move is constructed using a Code and two integers; the number of
+-- exact matches and the number of regular matches
+data Move = Move Code Int Int
+          deriving (Show, Eq)
+
+-- List containing all of the different Pegs
+colors :: [Peg]
+colors = [Red, Green, Blue, Yellow, Orange, Purple]
+
+-- Exercise 1 -----------------------------------------
+
+-- Get the number of exact matches between the actual code and the guess
+exactMatches :: Code -> Code -> Int
+
+exactMatches [] _ = 0
+exactMatches _ [] = 0
+exactMatches (x:xs) (x':xs') 
+    |x == x' = 1 + (exactMatches xs xs')
+    |otherwise = (exactMatches xs xs')
+
+-- Exercise 2 -----------------------------------------
+-- Write a function that returns the number of total matches between the secret code and the guess
+
+-- helper
+sumLists :: [Int] -> [Int] -> [Int]
+sumLists [] _ = []
+sumLists _ [] = []
+sumLists (x:xs) (x':xs') = (x + x'):(sumLists xs xs')
+
+countColors :: Code -> [Int]
+
+countColors [] = [0,0,0,0,0,0]
+countColors (x:xs)
+    | x == Red    = sumLists [1,0,0,0,0,0] (countColors xs)
+    | x == Green  = sumLists [0,1,0,0,0,0] (countColors xs)
+    | x == Blue   = sumLists [0,0,1,0,0,0] (countColors xs)
+    | x == Yellow = sumLists [0,0,0,1,0,0] (countColors xs)
+    | x == Orange = sumLists [0,0,0,0,1,0] (countColors xs)
+    | otherwise   = sumLists [0,0,0,0,0,1] (countColors xs)
+
+sumMin :: [Int] -> [Int] -> Int
+sumMin [] _ = 0
+sumMin _ [] = 0
+sumMin (x:xs) (x':xs') = (minimum [x, x']) + (sumMin xs xs')
```

Comment:
when comparing 2 elements it's better to use [min](https://hackage.haskell.org/package/base-4.15.0.0/docs/Prelude.html#v:min).  You could also avoid explicit recursion by using zipWith and sum here.